### ISSUE TITLE
[Regression] RN 0.76 stretches `Modal` to full screen on Android API < 30 if edge-to-edge mode is enabled

### ISSUE THREAD

**Issue by kirillzyusko** [facebook/react-native#47307]
### Description

If edge-to-edge mode is enabled for the app:

```kt
WindowCompat.setDecorFitsSystemWindows(this.window, false)
```

And Modal is shown, then we have a different behavior (see steps for reproduction and screenshots section).

Prior to RN 0.75 we had a consistent behavior and content was always shown above the navigation bar (independently from MainActivity configuration). I hope to get this consistent behavior in RN 0.76 as well. If we want to customize navigation bar behavior (and draw a content behind them) we can take a power of this PR: https://github.com/facebook/react-native/pull/47254

So keeping all information above in the head I tend to think that this is a regression 🙂 

### Steps to reproduce

1. Run reproduction example on API 29
2. Click on "Show modal" button
3. Check that content goes beyond navigation bar
4. Run reproduction example on API 30
5. Click on "Show modal" button
6. Check that the content is shown above the navigation bar

### React Native Version

0.76.1

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.6.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 148.58 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.14.0
    path: ~/.nvm/versions/node/v20.14.0/bin/node
  Yarn:
    version: 1.22.22
    path: ~/.nvm/versions/node/v20.14.0/bin/yarn
  npm:
    version: 10.7.0
    path: ~/.nvm/versions/node/v20.14.0/bin/npm
  Watchman:
    version: 2024.10.21.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.14.3
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12483815
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.9
    path: /usr/bin/javac
  Ruby:
    version: 2.7.6
    path: /Users/kirylziusko/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.0
    wanted: 15.0.0
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.1
    wanted: 0.76.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```

### Stacktrace or Logs

```text
There is no stacktrace (since it's not a crash). But let me know if you need any kind of logs.
```

### Reproducer

https://github.com/kirillzyusko/react-native-modal-edge-to-edge-issue

### Screenshots and Videos

|API 29|API 30|
|-------|------|
|<img width="509" alt="image" src="https://github.com/user-attachments/assets/86c85e7a-dd9f-4113-b5f7-fa3437957a9e">|<img width="505" alt="image" src="https://github.com/user-attachments/assets/9d5e7b5c-0c95-4bba-b894-b1e501a3e217">|

**Comment by JacksonHonoro** [comment]
I had a problem with react native modal after version 0.76.x. My modal without absolute position inside a view with flex: 1 was very displaced and even adjusting the layout the buttons inside it did not work... What solved this problem was I went to the **android/gradle.properties** file and disabled the new architecture **(newArchEnabled=false)**, after that the modal behaved correctly.
![code](https://github.com/user-attachments/assets/a30acea5-9e94-4aad-aaba-8b48a8cbac0c)

Then run:
```shell
yarn start --reset-cache
yarn android
```

**Comment by alanleedev** [comment]
@JacksonHonoro Could you provide screenshots and Reproducer ?

**Comment by alanleedev** [comment]
@kirillzyusko Could we close this or was there anything that still needs to be looked into?

**Comment by kirillzyusko** [comment]
@alanleedev yes, I believe it can be closed. I think this bug still can be reproduced if you use a different value of `statusBarTranslucent` (comparing to my example project). But I also think this bug can be fixed with addition of a new property called `navigationBarTranslucent`, so yes, let's close that issue!

**Comment by MauricioCarrion** [comment]
> I had a problem with react native modal after version 0.76.x. My modal without absolute position inside a view with flex: 1 was very displaced and even adjusting the layout the buttons inside it did not work... What solved this problem was I went to the **android/gradle.properties** file and disabled the new architecture **(newArchEnabled=false)**, after that the modal behaved correctly. ![code](https://github.com/user-attachments/assets/a30acea5-9e94-4aad-aaba-8b48a8cbac0c)
> 
> Then run:
> 
> yarn start --reset-cache
> yarn android

I've been passing by the exactly same problem, but that's "solution" worked for me

**Comment by scott-harrison** [comment]
I feel like im missing something. So the solution is to disable new arch for android. That sounds like a temp fix. 

Is there a real solution without turning new arch off.
